# Run this app locally
```
# install.packages("devtools")
devtools::install_github("sawsimeon/OSFP")
OSFP::run_app()
```
# Idea behind the app 

<p>According to the <a href="http://en.wikipedia.org/wiki/Bioluminescence">Wikipedia</a>, bioluminescence is the production and emission of light by a living organism. We have taken light for granted. Without light, we will have many problem doing daily jobs. And also, animals generally have a hard time getting around. Besides light from the <em>SUN</em>, we can use electricity to get a light source but for some animals, they can make their own light. That is called bioluminescence. It is a natural  light emission from living organism. Animals use bioluminescence for many reasons. Fireflies use it to attract mates. Cookiecutter shark use it to attract its prey. And Jelly fish bioluminescence to protect itself from predator. From drug discoveries to monitoring of analyses, the light source from animals has become essential tools for biomedical and live cell imaginings. Of all the bioluminescence proteins that are discovered and isolated, which are luciferases, green fluorescent proteins and photoproteins, green fluorescent protein is the fairest of them all. </p>

<h1>
<a id="history-of-green-fluorescent-protein" class="anchor" href="#history-of-green-fluorescent-protein" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>History of Green Fluorescent Protein</h1>

<p>It was in 1960s that a scientist named Osama shimamura discovery the Green Fluorescent Protein. In ofter to purify protein, he needs to harvest many thousand of jelly fish and carefully purified the protein responsible for the green luminesence. He did this and he actually found 2 proteins. One was the Aqurin, an enzyme that generate blue light when it bid to calcium. The other protein was able to absorb the blue light and reemit the green light. The second protein was of course, the green fluorescent protein. Although, GFP was discovered back in 1960s, the research into the properties and characterization of GFP processed very slowly. In 1992, it was reported that the gene encoding GFP has bee cloned by Prasher, What cloning gets you is a complete chemical understanding of the protein. Because, once you know the DNA sequence, you also know the protein sequence.
Two years after that in 1994, came the first report that said you can take this DNA and put it in the other organism (bacteria or worms) and this DNA could still be produced protein and that protein can turn green fluorescent. It was reported by Chafie. This was really an important step in the fluorescent protein development because this was the first time to show that GFP donâ€™t need anything to fluorescent. It was self-sufficient to form green fluorescent. Two years after that, the three dimensional structure of GFP was report by two groups, and eventually, in 2008, Shimamura, Martin Chafie and Roger Tisen were awarded noble prize for their discovery and development of green fluorescent protein. <img src="http://2.bp.blogspot.com/_fmViYbhmOd0/TKjgToUFnsI/AAAAAAAAAD0/jggFP963keU/s1600/GFP_slate1.png" alt="3-Dimensional Structure of Green Fluorescent Protein"></p>

<h1>
<a id="why-green-fluorescent-protein-for-tagging-proteins" class="anchor" href="#why-green-fluorescent-protein-for-tagging-proteins" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Why Green Fluorescent Protein for tagging proteins</h1>

<p>Sometimes , Proteins are labeled with fluorescent tag. so we can monitor their the movement and localization. Normally, proteins are  tagged with organic fluorophores. An example is  a very popular fluophore called flourcines. Fluorresceins can convanlently bind to the amino groups of protein. But the problem is that their biniding are not very specific because all amino acids have amino groups. And on top of that , you also need to carefully repurified the protein again.</p>

<p>Another way to label protein is to use molecular biology. This approach is very good. The way is to tag the gene of the interest with a gene of fluorescent protein. This gene will still be transcribed and translated into a protein. So we can get ready-made protein that already has fluorescent label. And one of the popular  protein is Luciferase. This protein is  hilghly fluroescent  but the limitation of this protein is that it needs an substrate and cofactor.</p>

<p>So GFP becomes a very promising choice because it is self sufficient to form a chromophore. It does not need any cofactor or enzyems. It has shown that GFP can be expressed alive living organisms, including E coli, C. Elegans, yeast and flies and it is brightly fluorescent. 
</p>

<h1>
<a id="why-computational-studies-instead-of-traditional-experiments" class="anchor" href="#why-computational-studies-instead-of-traditional-experiments" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Why computational studies instead of traditional experiments</h1>

<p>Strictly specking, it is because computational studies save money, time and energy. We were the first group to applied predictive modelling for green fluorescent proteins back in a decade ago. The paper was submitted two years earlier but got accepted in <a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-016-0185-8">2016</a>. Now, in the spirit of open science and reproducible research, we applied predictive modelling for green fluorescent protein without spending a single cent. From descriptor calculation to statical assessment to the journal publisher, everything is transparent. Please have a look at the preprinted <a href="https://peerj.com/preprints/922/">version</a> </p>
<p>Having trouble with Pages? Please contact <a href="https://github.com/sawsimeon">Saw Simeon</a></p>

<a href="https://github.com/sawsimeon/OliFP">Fluorescent Proteins</a> is maintained by <a href="https://github.com/sawsimeon">sawsimeon</a>.</span>
